### DISCUSSION TITLE
separate sync with block chain from sync to restock seeds, resources ..

### DISCUSSION THREAD

**Discussion by dragonhunter2017** [sunflower-land/sunflower-land#2178] [Ideas]
Recently the polygon gas fee is significantly larger than the game fee . Lots of us don't wait one week to sync, rather sync everyday. Now high gas fee is costing us a lot. If technically possible, my proposal is

make two separate sync 
1. sync to re-stock
2. sync to secure resources to block chain (can be forced to do after 5 re-stock sync)

**Comment by rasheed19871** [comment]
that is true most of my friends can't afford to sync everyday espacially for tools (wood)

**Comment by adamhannigan** [comment]
@dragonhunter2017 Agreed, the team is investigating a few possible solutions to the high gas fees.

One possibility, is buying 'SFL Tickets' in bulk (e.g. 10 at a time) in a single transaction for $1.

Players can then use these tickets for instant seed restocks, instead of needing to submit multiple transactions. What are your thoughts on that idea?

**Reply by dragonhunter2017** [reply]
> One possibility, is buying 'SFL Tickets' in bulk (e.g. 10 at a time) in a single transaction for $1.
> 
> Players can then use these tickets for instant seed restocks, instead of needing to submit multiple transactions. What are your thoughts on that idea?

I think , it is very good idea.

**Reply by Polysemouse** [reply]
@adamhannigan I really, really need this.  I (and probably other players) slowed down the syncing rate since the release of on chain bumpkin stats (because more data per sync = more gas fees).  Oftentimes the gas fees cost more than the $0.1 dev base fees.  The gas fees are wasted and are not going to the devs.

Buying $1 tickets allow players to sync less and effectively paying ~90% less gas fees per sync, while still ensuring players will sync every 10 tickets are used so the on chain records are still somewhat updated.

The blockchain sync limits would need to be updated.  Some limits are too low currently and are hindering gameplay for players with a lot of boosts, but that's another topic to discuss (we still need the blockchain limit to mitigate the damage if players find a way to mint 11 octodecillion iron pickaxes).

**Reply by poteitouweb3** [reply]
Reestock for seeds and tools or seeds only? Would be awesome for both @adamhannigan

**Reply by caiofelipegq** [reply]
The tickets its a really good idea, but i thinks they need restore the tools too. I believe the most numbers of sync are for replace the axes.

**Reply by WWW-Spaceships** [reply]
I hope it's seeds and tools because tools are more important to me.

**Reply by malkaron** [reply]
As long as these tickets are also valid for the items purchased at the farm to appear in the goblin village, it's fine. Is that possible?

**Comment by Danverse98** [comment]
I really like the idea of introducing SFL Tickets as a form to hoard some syncs beforehand when the gas is low.

One thing i was thinking about lately is some form of an ''SFL Pass'' that would remove the need to sync as a whole for around 10 dollars in Matic per month, plus it could give some bonus perks and remove the hoarder limit to make people interested in getting one. I know it sort of copies the usual form of other mainstream games but i'm sure that would help both the devs and ease the brains of players constantly doing the math with their syncs.

This system is also amazing to keep the players engaged because they now have a whole month to fully utilize their pass.

10 dollars is around 100 syncs which only the best hoarders barely hit each month, only difference is that we currently pay 20 cents on top of each sync to Polygon so our 10 dollars is suddenly 30 for little to no reason and ends up in Polygons pockets regardless.

Just adding my 2 cents to the discussion, SFL Tickets right now are probably the best, quick solution to the whole thing.

**Comment by cryptokohai** [comment]
Paying gas, syncing frequently, helps to develop the blockchain. 

Taking stuff off-chain will distance ourselves from blockchain. 

SFF was one extreme, where every action was synced on-chain

The other extreme, which is to sync every 10000 restock, at which point you might as well become an ordinary web2 game. 

I believe the answer lies somewhere in the middle, but bearing in mind to stay part of, and continue to contribute to, the blockchain.

Which is more important? 

Having 100 SFL on servers, or; having 100 SFL on-chain?

There is no right answer.

You cant play the game without item/seeds/currency on servers, yet you don’t own item/seeds/currency if it isnt on-chain. 

But since everyone is arguing for gas-less direct-to-server restocks, bypassing polygon (for extended periods of time), I chose to speak for the other side of things.

Adapting to the dynamics of the blockchain is part of being on the blockchain. 

When gas is high, players should adapt their playstyle, such as longer duration between syncs, or more frequent syncs to build up inventory when gas is low. 

This means less revenue for devs when gas is high, likewise, devs should adapt to this. For example, by increasing player base and NFT value/trade frequency, so that there is an alternate (and maybe more lucrative) source of revenue when gas is high and players avoid sync.

**Reply by dragonhunter2017** [reply]
> Paying gas, syncing frequently, helps to develop the blockchain.
> 
> Taking stuff off-chain will distance ourselves from blockchain.
> 
> SFF was one extreme, where every action was synced on-chain
> 
> The other extreme, which is to sync every 10000 restock, at which point you might as well become an ordinary web2 game.
> 
> I believe the answer lies somewhere in the middle, but bearing in mind to stay part of, and continue to contribute to, the blockchain.
> 
> Which is more important?
> 
> Having 100 SFL on servers, or; having 100 SFL on-chain?
> 
> There is no right answer.
> 
> You cant play the game without item/seeds/currency on servers, yet you don’t own item/seeds/currency if it isnt on-chain.
> 
> But since everyone is arguing for gas-less direct-to-server restocks, bypassing polygon (for extended periods of time), I chose to speak for the other side of things.
> 
> Adapting to the dynamics of the blockchain is part of being on the blockchain.
> 
> When gas is high, players should adapt their playstyle, such as longer duration between syncs, or more frequent syncs to build up inventory when gas is low.
> 
> This means less revenue for devs when gas is high, likewise, devs should adapt to this. For example, by increasing player base and NFT value/trade frequency, so that there is an alternate (and maybe more lucrative) source of revenue when gas is high and players avoid sync.

Hi, you have to understand why gas fee is high sometimes. Every blockchain can handle maximum transaction per second (Polygon has the potential to handle up to 65,000 transactions per second, whereas Ethereum can process only up to roughly 17 transactions per second. ) when transctions reach close to critical, fees goes to roof to prevent the blockchain not to be clogged. 

No one is telling here to go off chain. Lots of players sync several times everyday, when gas fee is high (it means the number of transaction in polygon network is close to maximum and it can't handle anymore transaction), it is very costly for them to sync, sometimes it may not be possible to sync as network is clogged.

if we are required to sync less, it will help the blockchain, on the contrary. in future, if when we will have more player, then we need to implement way to sync less as polygon can't handle large amount of transaction.

for more info, please check this article, 
https://cointelegraph.com/blockchain-for-beginners/polygon-blockchain-explained-a-beginners-guide-to-matic#:~:text=Polygon%20has%20the%20potential%20to,fees%20to%20users%20for%20pennies.

https://cryptobriefing.com/nft-gamers-are-clogging-up-polygon/

**Reply by WWW-Spaceships** [reply]
Well said, I appreciate the last guys thought process as well.
